utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/source/UniversalKafkaSourceFactory.scala

```diff
@@ -122,7 +134,48 @@ class UniversalKafkaSourceFactory[K: ClassTag: NotNothing, V: ClassTag: NotNothi
     val formatterSchema = schemaBasedMessagesSerdeProvider.deserializationSchemaFactory.create[K, V](kafkaConfig, None, None)
     val recordFormatter = schemaBasedMessagesSerdeProvider.recordFormatterFactory.create[K, V](kafkaConfig, formatterSchema)
 
-    implProvider.createSource(params, dependencies, finalState.get, List(preparedTopic), kafkaConfig, deserializationSchema, recordFormatter, kafkaContextInitializer)
+    implProvider.createSource(params, dependencies, finalState.get, List(preparedTopic), kafkaConfig, deserializationSchema,
+      recordFormatter, kafkaContextInitializer, getParameters(preparedTopic, versionOption))
+  }
+
+  private def getParameters(topic: PreparedKafkaTopic, version: SchemaVersionOption)(implicit nodeId: NodeId): KafkaTestParametersInfo = {
+    getSchema(topic, version)
+      .andThen(handleSchemaType)
+      .valueOr(e => throw new RuntimeException(e.toList.mkString("")))
+  }
+
+  private def getSchema(topic: PreparedKafkaTopic, version: SchemaVersionOption)(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, RuntimeSchemaData[ParsedSchema]] = {
```

Comment:
We already have this schema in `UniversalKafkaSourceFactoryState.valueRuntimeSchema`. Haven't we?

Reply:
Yeah - I've reinvented a wheel here... Removed `getSchema` and used `valueRuntimeSchema`. Thanks for that :)